### PR TITLE
feat(server): fallback option

### LINKED ISSUE #1863: nuxtServerInit triggers twice
```js
nuxtServerInit({ dispatch, commit  }, { req, isServer }) {
    if (req.session && req.session.access_token) {
      if (isServer) {
        return dispatch('user', req.session.access_token)
    } else {
      delete req.session.access_token
      return commit('destroy')
    }
  }
}
```
this is triggering the user action twice ... 

note: the user action is returning a promise.

<div align="right"><sub><em>This bug report is available on <a href="https://nuxtjs.cmty.io">Nuxt.js</a> community (<a href="https://nuxtjs.cmty.io/nuxt/nuxt.js/issues/c1674">#c1674</a>)</em></sub></div>

### LINKED ISSUE #4016: NuxtserverInit and Middleware are being hit on *.css.map requests
### Version

[v2.0.0](https://github.com/nuxt.js/releases/tag/v2.0.0)

### Steps to reproduce

I noticed that NuxtserverInit and Middleware are being called on every  *.css.map request. 

Probably, to reprouduce one needs to   set  extractCSS: true,

### What is expected ?

NuxtserverInit should only be called on pages

### What is actually happening?

*.css.map requests should not hit middleware

### Additional comments?

This is causing a serious performance hit

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c7871">#c7871</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by pi0**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Resolves #4016, resolves #1863

Options kept in `renderer` key for consistency with our current conventions. We should eventually move all server related options into `server.js` (Mentioned in TODO)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly. (PR: nuxt/docs#982)
- [x] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

File: packages/config/src/config/render.js [hunk 1]
```diff
@@ -23,5 +25,16 @@ export default () => ({
     index: false,
     // 1 year in production
     maxAge: '1y'
+  },
+  // https://github.com/nuxt/serve-placeholder
+  // Options for /_nuxt/*
+  distPlaceholder: {},
```
Comment (clarkdo) [inline packages/config/src/config/render.js]: How about:

```js
placeholders: {
  static: {},
  dist: {}
}
```

**Review by clarkdo — DISMISSED** [review]

File: packages/config/src/config/render.js (hunk 1 above)
Comment (pi0) [reply packages/config/src/config/render.js]: @clarkdo good idea. will change it.

**Review by aldarund — DISMISSED** [review]

**Comment by pi0** [comment]
@clarkdo @aldarund Sorry for change after review :D What do you think about renamed `fallback` option? I thought it may be more general and meaningful for this purpose.

**Review by clarkdo — APPROVED** [review]

**Review by TheAlexLichter — APPROVED** [review]

**Comment by aldarund** [comment]
LGTM

**Pull request merged**